src/media/local-audio-track.ts

```diff
@@ -0,0 +1,89 @@
+import { logger } from '../util/logger';
+import { LocalTrack } from './local-track';
+
+export type AudioDeviceConstraints = {
+  deviceId?: string;
+};
+
+export type StaticAudioEncoderConfig = {
+  low: '16kHz';
+};
+
+export type AudioEncoderConfig = {
+  bitrate?: number;
+  sampleRate?: number;
+  sampleSize?: number;
+  channelCount?: number;
+};
+
+export type MicrophoneConstraints = {
+  echoCancellation?: boolean;
+  autoGainControl?: boolean;
+  noiseSuppression?: boolean;
+  microphoneDeviceId?: string;
+  encoderConfig?: AudioEncoderConfig;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TrackEffect = any;
+/**
+ * Represents a audio track for a audio source.
+ */
+export class LocalAudioTrack extends LocalTrack {
+  private effects: Map<string, TrackEffect> = new Map();
+
+  /**
+   * Get an effect by name.
+   *
+   * @param name - The effect name.
+   * @returns A MicrophoneEffect.
+   */
+  getEffect(name: string): TrackEffect | undefined {
+    const effect = this.effects.get(name);
+
+    if (!effect) {
+      logger.log(`No effect found with name '${name}'`);
+    }
+
+    return effect;
+  }
+
+  /**
+   * Adds an effect to a local track.
+   *
+   * @param name - The name of the effect.
+   * @param effect - The effect to add.
+   */
+  async addEffect(name: string, effect: TrackEffect): Promise<void> {
+    await effect.load(new MediaStream([this.getMediaStreamTrack()]));
+    this.setMediaStreamTrackWithEffects(effect.getUnderlyingStream().getAudioTracks()[0]);
+    this.effects.set(name, effect);
+  }
+
+  /**
+   * Cleanup local microphone track.
+   */
+  disposeEffects(): void {
+    this.effects.forEach((effect: TrackEffect) => effect.dispose());
+  }
+
+  /**
+   * Adds background noise cancellation effect.
+   *
+   * @param name - The name of the effect.
+   * @param effect - The effect to add.
+   */
+  async addBNREffect(name: string, effect: TrackEffect): Promise<void> {
```

Comment:
why do we need an effect specific function like this? Each effect should just implement the same TrackEffect interface and then can be used with addEffect(), right? Otherwise you will have to keep adding a new function every time you add an effect and effects are not even in the same repo, right? and it's just duplicated code and you will have to check if the right effect instance is added, etc, It feels like it just creates more code to maintain and test but no added value.

Reply:
The only reason we added this one was avoid the name it just encloses  the name , ok

Reply:
removed